app/scripts/directives/service-definition-edit.js

```diff
@@ -46,6 +46,9 @@
             // if sequence type keys have string value, convert them to array
             svalue = convertSeqKeyValueToArray(svalue, skey);
 
+            // if environment key values are a sequence, convert them to a hash array
+            svalue = convertEnvironmentSeqOrHashToArray(svalue, skey);
```

Comment:
Firstly, I like to keep the functions independent in what they do, so I like to include all checks inside the function. Secondly, I can't do an if/else, as both the methods apply to 'environment'. The ```convertSeqKeyValueToArray(svalue, skey)``` looks for just string values for possible sequence type keys. The 'environment' key being hybrid falls in that category as well. 

If you insist, I can at the most do this:

```
            if (isKeyTypeSequence(skey)){
              // if sequence type keys have string value, convert them to array
              svalue = convertSeqKeyValueToArray(svalue, skey);
            }
            if (skey === 'environment') {
              // if environment key values are a sequence, convert them to a hash array
              svalue = convertEnvironmentSeqOrHashToArray(svalue, skey);
            }
```